include/deal.II/matrix_free/fe_point_evaluation.h

```diff
@@ -68,7 +68,10 @@ namespace internal
       using scalar_value_type = Tensor<1, n_components, ScalarNumber>;
       using vectorized_value_type =
         Tensor<1, n_components, VectorizedArrayType>;
-      using gradient_type = Tensor<1, n_components, Tensor<1, dim, Number>>;
+      using unit_gradient_type =
+        Tensor<1, n_components, Tensor<1, dim, Number>>;
+      using gradient_type =
+        Tensor<1, n_components, Tensor<1, spacedim, Number>>;
       using scalar_gradient_type =
         Tensor<1, n_components, Tensor<1, dim, ScalarNumber>>;
       using vectorized_gradient_type =
```

Comment:
I opted to keep the names without `_unit` for the parts involving `dim` and choose `real_gradient_type` for the space-dim gradient. I admit that in the user interface I have to define `FEPointEvaluation::gradient_type` to be an alias for `real_gradient_type`, but I think all the names are in the internal namespace and most compact this way. One could interpret them as being the user-facing interface in `spacedim` and the evaluator-facing interface in `dim` (which is exactly what we are doing). What do you think?

Reply:
Honestly, I think it would be better to be explicit and call every `dim` gradient `_unit_` and every `spacedim` gradient `_real_`. Even if it is a bit lengthier, it forces us to be consistent.